datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,397 @@
+    for name in h.multi_split(names, [" | "]):
+        person.add("name", name)
+    person.add("citizenship", data.pop("Citizenship", None))
+    person.add("taxNumber", data.pop("Tax Number", None))
+    person.add("sourceUrl", data.pop("Links").split(" | "))
+    archive_links = data.pop("Archive links", None)
+    if archive_links is not None:
+        for archive_link in archive_links.split(" | "):
+            person.add("sourceUrl", archive_link)
+    dob_pob = data.pop("Date and place of birth", None)
+    if dob_pob:
+        dp_parts = dob_pob.split(" | ")
+        # If we get more than one part, unpack it into dob and pob
+        if len(dp_parts) == 2:
+            dob, pob = dp_parts
+            h.apply_date(person, "birthDate", dob)
+            person.add("birthPlace", pob)
+        elif len(dp_parts) == 1:
+            # If there’s only one part, we assume it's just the dob
+            dob = dp_parts[0]
+            h.apply_date(person, "birthDate", dob)
+    if positions:
+        for pos_parts in h.multi_split(positions, [" | ", " / "]):
+            person.add("position", pos_parts)
+    person.add("topics", "poi")
+
+    sanction = h.make_sanction(context, person)
+    sanction.add("reason", data.pop("Reasons").replace(" | ", " "))
+    sanction.add("sourceUrl", link)
+
+    context.emit(person, target=True)
+    context.emit(sanction)
+    context.audit_data(data)
+
+
+def crawl_company(context: Context, details_container, link):
+    data = {}
+    for row in details_container.findall(".//div[@class='row']"):
+        label_elem = row.find(".//div[@class='col-12 col-sm-4 yellow']")
+        value_elem = row.find(".//div[@class='col-12 col-sm-8']")
+
+        if label_elem is not None and value_elem is not None:
+            data = extract_label_value_pair(label_elem, value_elem, data=data)
+
+    name = data.pop("Name")
+    name_abbr = data.pop("Abbreviated name of the legal entity", None)
+    reg_num = data.pop("Registration number")
+
+    company = context.make("Company")
+    company.id = context.make_id(name, name_abbr, reg_num)
+    company.add("name", name)
+    company.add("name", name_abbr)
+    company.add("registrationNumber", reg_num)
+    company.add("address", data.pop("Address"))
+    company.add("country", data.pop("Country"))
+    company.add("taxNumber", data.pop("Tax Number"))
+    company.add("sourceUrl", data.pop("Links").split(" | "))
+    archive_links = data.pop("Archive links", None)
+    if archive_links is not None:
+        for archive_link in archive_links.split(" | "):
+            company.add("sourceUrl", archive_link)
+
+    company.add("topics", "poi")
+    sanction = h.make_sanction(context, company)
+    sanction.add("reason", data.pop("Reasons"))
+    sanction.add("sourceUrl", link)
+
+    context.emit(company, target=True)
+    context.emit(sanction)
+    context.audit_data(data)
+
+
+def extract_next_page_url(doc, base_url, next_xpath):
+    doc.make_links_absolute(base_url)
```

Comment:
let's do this right after we fetch the page. We want absolute links for the rest of the process too, and that'll be consistent with other crawlers.